jaraco/develop/projects-run.py

```diff
@@ -2,40 +2,37 @@
 Routine to run a command across all projects.
 """
 
-import argparse
+import functools
 import subprocess
+from typing import List, cast
 
-import autocommand
+import typer
+from jaraco.ui.main import main
+from typing_extensions import Annotated
 
 from . import filters, git
 
-parser = argparse.ArgumentParser()
-parser.add_argument(
-    '--keyword',
-    '-k',
-    dest='selectors',
-    type=filters.Keyword,
-    default=filters.Selectors(),
-    action='append',
-)
-parser.add_argument(
-    '--tag',
-    '-t',
-    dest='selectors',
-    type=filters.Tag,
-    default=filters.Selectors(),
-    action='append',
-)
-parser.add_argument('args', nargs='*')
 
-
-@autocommand.autocommand(__name__, parser=parser)
-def main(
-    selectors: filters.Selectors,
-    args=None,
+@functools.partial(
+    main,
+    app=typer.Typer(
+        context_settings=dict(allow_extra_args=True, ignore_unknown_options=True)
+    ),
+)
+def run(
+    tag: Annotated[
+        List[filters.Tag], typer.Option('--tag', '-t', parser=filters.Tag)
+    ] = [],
+    keyword: Annotated[
+        List[filters.Keyword],
+        typer.Option('--keyword', '-k', parser=filters.Keyword),
+    ] = [],
+    args: typer.Context = typer.Option(None),
 ):
+    cmd = cast(List[str], args.args)
```

Comment:
That doesn't look correct. `typer.Context` is always injected when there's an annotation that inherits from `click.Context` (see [inference implementation](https://github.com/fastapi/typer/blob/76ca3e3397f13993cae4a9d68cc60e51c93253d8/typer/main.py#L543-L559)).

```suggestion
    args: typer.Context,
):
    cmd = cast(List[str], args.args)
```

Additionally, I'd name the context parameter `context` or `ctx`. `args.args` reads weird. But that's just an opinion.

Reply:
Good call. Not sure how I missed this in the review. I went through the whole thing in the files dialog and didn't see it. I'll get it after the fact.

Reply:
I tried removing `= typer.Option(None)`, but you can't do that because `tag` and `keyword` have default values. Maybe it would work if cmd were keyword-only.
```suggestion
    *,
    args: typer.Context,
):
    cmd = cast(List[str], args.args)
```

Reply:
Yes, that works; done in 3a80b14.